pylhc/BSRT_logger.py

```diff
@@ -0,0 +1,100 @@
+'''
+BSRT logger
+-------------------
+
+Script used during RunII to log detailed BSRT data and save it for later analysis.
+Data from the BSRT for each timestep is put in dictionary and append to a list.
+The list is then saved as a pickle and gzipped.
+Proper testing requires communication with FESA class, possible only from the TN.
+
+Original authors: E. H. Maclean, T. Persson, G. Trad
+'''
+import sys
+import os
+import time
+import datetime as dt
+import pickle
+import pyjapc
+#export PATH="/user/bdisoft/operational/bin/Python/PRO/bin:$PATH
+
+##########################################
+
+
+def parse_timestamp(thistime):
+    accepted_time_input_format = ['%Y-%m-%d %H:%M:%S.%f',
+                                  '%Y-%m-%d %H:%M:%S',
+                                  '%Y-%m-%d_%H:%M:%S.%f',
+                                  '%Y-%m-%d_%H:%M:%S',
+                                  '%Y/%m/%d %H:%M:%S.%f']
+    for fmat in accepted_time_input_format:
+        try:
+            dtobject = dt.datetime.strptime(thistime, fmat)
+            return dtobject
+        except ValueError:
+            pass
+    timefmatstring = ''
+    for fmat in accepted_time_input_format:
+        timefmatstring = timefmatstring+'\"'+fmat+'\" ,   '
+    sys.tracebacklimit = 0
+    raise ValueError(
+        'No appropriate input format found for start time of scan (-s).\n ---> Accepted input formats are:   '+timefmatstring)
+##########################################
+
+
+# function to help write output from datetime objects in standard format throughout code
+def convert_to_data_output_format(dtobject):
+    output_timestamp = dtobject.strftime('%Y-%m-%d-%H-%M-%S.%f')
```

Comment:
Why not the CERN format with the @?

Reply:
script was uploaded as was since testing etc. not possible . Once run III this should be tested again and possibly refactored. For the time being it has been replaced and bsrt analysis was updated to use both formats.